Refactor FirstLesson component: Make calendar functions working and m…

chimpiriudaykiran opened the pull request:
Make calendar functions working and made lessons link active

- Defined a dynamic `EventDetails` type for calendar events.
- Created `handleAddToCalendar` and `handleDownloadICS` functions to accept dynamic event and add to calendar.
- Added `event` object containing the details of the live session.
- Integrated a dropdown menu for "Add to Calendar" with options for:
  - Google Calendar (calls `handleAddToCalendar(event)`)
  - Download ICS (calls `handleDownloadICS(event)`)
- Made lessons button clickable.

weic6 commented:
could you rebase this branch onto our latest main branch `develop`?

weic6 reviewed (commented):
These functions are executing on render instead of on click. The issue is 
`onClick={handleDownloadICS(event)}` immediately invokes the function. please fix it.

chimpiriudaykiran commented on src/app/(authorized)/dashboard/components/FirstLesson.tsx:
```diff
@@ -2,35 +2,131 @@ import Button3 from '@/components/atoms/Button3'
+                                  <li>
+                                      <Button3
+                                          text='Download ICS'
+                                          onClick={handleDownloadICS(event)}
```
Fixed the function execution by passing the function reference.

chimpiriudaykiran commented on src/app/(authorized)/dashboard/components/FirstLesson.tsx:
```diff
@@ -2,35 +2,131 @@ import Button3 from '@/components/atoms/Button3'
+                                  <li>
+                                      <Button3
+                                          text='Google Calendar'
+                                          onClick={handleAddToCalendar(event)}
```
Fixed the function execution by passing the function reference.

weic6 reviewed (commented):
Nice, both Google Calendar Integration and ics file works for me. @saivivek116 i think this PR is ready to merge upon your final review.

chimpiriudaykiran commented:
Updated the changes as requested. 

Replaced onMouseEnter and onMouseLeave events with an onClick toggle for the dropdown menu.
Added arrow direction logic to dynamically switch between "▼" (collapsed) and "▲" (expanded).
Ensures the dropdown opens only on button click and closes when clicked again.
